tensorrt_llm/_torch/modules/fused_moe/interface.py

```diff
@@ -64,6 +144,19 @@ def __init__(
         self.swiglu_alpha = swiglu_alpha
         self.swiglu_beta = swiglu_beta
         self.swiglu_limit = swiglu_limit
+        self.layer_idx = layer_idx
+        self.layer_idx_str = str(layer_idx) if layer_idx is not None else None
+
+        self.register_to_config = False
+        # We only register TRTLLM attention layers to config.
+        if model_config is not None and self.layer_idx_str is not None:
+            if "moe_layers" not in model_config.extra_attrs:
+                model_config.extra_attrs["moe_layers"] = {}
+            assert self.layer_idx_str not in model_config.extra_attrs["moe_layers"], \
+                f"Duplicate MoE layer for layer_idx={self.layer_idx_str}"
+            model_config.extra_attrs["moe_layers"][
+                self.layer_idx_str] = weakref.ref(self)
```

Comment:
Why do you need to convert `layer_idx` from int to string? int can be used as hash keys as well.